### DISCUSSION TITLE
Formik is not submitting values. I am showing 2 out of 5 fields on Sign In and 5 fields on SignUp. formik is asking to  also fill other 3 fields which I am not showin on Sign In.

### DISCUSSION THREAD

**Discussion by ans4asif** [jaredpalmer/formik#3101] [Q&A]
I am using one component as Sign In and Sign Up form. I have Fields of FirstName, LastName , Email and Password. I have also applied Yup validation and set the "required" property on every field. As I am showing only two fields during Sign In so the formik form is not submitting the form values and asking to fill the required form fields. So, do I have to create another component for SignIn Form or can be done in this single form.
 
`<Formik
          initialValues={initialValues}
          validationSchema={validationSchema}
          onSubmit={(values)=>{
            setFormData(values)
            console.log(formData);
          }}
        >
          {(formik) => (
            <Form className={classes.form} onSubmit={formik.handleSubmit}>
              <Grid container spacing={2}>
                {isSignUp && (
                  <>
                    <Input
                      name="firstName"
                      label="First Name"
                      type="text"
                      half
                    />
                    <Input name="lastName" label="Last Name" type="text" half />
                  </>
                )}`

                `<Input name="email" label="Email" type="email" />
                <Input
                  name="password"
                  label="Password"
                  type={showPassword ? "text" : "password"}
                  handleShowPassword={handleShowPassword}
                />
                {isSignUp ? (
                  <Input
                    name="confirmPassword"
                    type="password"
                    label="Confirm Password"
                  />
                ) : null}
              </Grid>
              <Button className={classes.submit}
                type="submit"
                variant="contained"
                fullWidth
                color="primary"
              >
                {isSignUp ? "Sign Up" : "Sign In"}
              </Button>
             
              <Grid container justify="flex-end">
               <Grid item>
              <Button color="primary" onClick={switchMode}>{isSignUp ?"Already have an account?Sign In":"Don't have an account?Sign Up"}
</Button>
              </Grid>
              </Grid>
            </Form>
            
          )}
        </Formik>
`

**Comment by johnrom** [comment]
@ans4asif you should provide a different validationSchema depending on whether your form is sign up or sign in.

I wouldn't mix these two forms personally, even though they are very similar I would make the child components reusable and implement two different forms. I would create a `PasswordField`, `ConfirmPasswordField`, etc, and then make the forms look like:

```jsx
// switchMode should switch components
const SignIn = (switchMode) => (
    <Formik
        initialValues={signInInitialValues}
        validationSchema={signInValidationSchema}
        onSubmit={(values)=> signIn(values)}
    >
        <Form className={classes.form}>
            <Grid container spacing={2}>
                <Input name="email" label="Email" type="email" />
                <PasswordField />
            </Grid>
            <SubmitButton>Sign In</SubmitButton>
            <Grid container justify="flex-end">
                <Grid item>
                    <Button color="primary" onClick={switchMode}>
                        Don't have an account? Sign Up
                    </Button>
                </Grid>
            </Grid>
        </Form>
    </Formik>
);

const SignUp = (switchMode) => (
    <Formik
        initialValues={signUpInitialValues}
        validationSchema={signUpValidationSchema}
        onSubmit={(values)=> signUp(values)}
    >
        <Form className={classes.form}>
            <Grid container spacing={2}>
                <Input name="email" label="Email" type="email" />
                <Input
                    name="firstName"
                    label="First Name"
                    type="text"
                    half
                />
                <Input name="lastName" label="Last Name" type="text" half />
                <PasswordField />
                <ConfirmPasswordField />
            </Grid>
            <SubmitButton>Sign Up</SubmitButton>
            
            <Grid container justify="flex-end">
                <Grid item>
                    <Button color="primary" onClick={switchMode}>
                        Don't have an account? Sign Up
                    </Button>
                </Grid>
            </Grid>
        </Form>
    </Formik>
);
```

**Reply by ans4asif** [reply]
Thanks, I got it.
